### ISSUE TITLE
Lines starting from 0

### ISSUE THREAD

**Issue by markusmoenig** [mchakravarty/CodeEditorView#113]
Hi,

would it be possible to add an option to start lines from 0 instead of 1 ? 

I am working on a chip emulator in assembly which has offsets into tags starting from 0.

Thanks :)

**Comment by mchakravarty** [comment]
@markusmoenig Sorry, but I'm not sure I understand the feature that you need. Do you need the first line to have a line number, which is labeled "0", or do you need the internal line count (in the line map) to start at 0?

Changing the line number that is printed to the left of each line would instead require a change, but a very simple one. 

Internally, the line map already starts from 0 and the API allows to use both 0-based or 1-based line numbers when creating a position value.

**Comment by markusmoenig** [comment]
I would need the line numbers to count up from 0. Is that already in the API ? Did not see it. 

Thankd

**Comment by markusmoenig** [comment]
@mchakravarty  While I have your attention, how do get the current CURSOR position when it changes ? I seem to only be able to access selections and scroll positions ?

I would need to get the current line position to show help information about the assembly instruction on that line.

Thanks

**Comment by mchakravarty** [comment]
> I would need the line numbers to count up from 0. Is that already in the API ? Did not see it.

No, that's not yet supported, but easy to add. I'll try to have a look at it this or next week. (Or feel free to have a go yourself and send a PR.)

**Comment by mchakravarty** [comment]
> While I have your attention, how do I get the current CURSOR position when it changes ? I seem to only be able to access selections and scroll positions ?

You can do this with the `setInfo` argument, which I added recently — see https://github.com/mchakravarty/CodeEditorView/blob/31c87f788bf948f158e9e22f0e537796b0566dd9/Sources/CodeEditorView/CodeEditor.swift#L209
